core/bazooka.go

```diff
@@ -337,6 +337,90 @@ func (b *Bazooka) DecodeState(stateBytes []byte) (ID, balance, nonce, token *big
 // Transactions
 //
 
+// SubmitBatch submits the batch on chain with updated root and compressed transactions
+func (b *Bazooka) SubmitBatch(commitments []Commitment) error {
+	b.log.Info(
+		"Attempting to submit a new batch",
+		"NumOfCommitments",
+		len(commitments),
+	)
+
+	if len(commitments) == 0 {
+		b.log.Info("No transactions to submit, waiting....")
+		return nil
+	}
+
+	var txs [][]byte
+	var updatedRoots [][32]byte
+	var aggregatedSig [][2]*big.Int
+	var totalTxs int
+
+	for _, commitment := range commitments {
+		compressedTxs, err := b.CompressTxs(commitment.Txs)
+		if err != nil {
+			b.log.Error("Unable to compress txs", "error", err)
+			return err
+		}
+		txs = append(txs, compressedTxs)
+		updatedRoots = append(updatedRoots, commitment.UpdatedRoot)
+		totalTxs += len(commitment.Txs)
+
+		// TODO cleanup
+		sig1 := commitment.AggregatedSignature[0:32]
+		sig2 := commitment.AggregatedSignature[32:64]
+		sig1bigInt := big.NewInt(0)
+		sig1bigInt.SetBytes(sig1)
+		sig2bigInt := big.NewInt(0)
+		sig2bigInt.SetBytes(sig2)
+		aggregatedSigBigInt := [2]*big.Int{sig1bigInt, sig2bigInt}
+		aggregatedSig = append(aggregatedSig, aggregatedSigBigInt)
+	}
+
+	b.log.Info("Batch prepared", "totalTransactions", totalTxs)
+
+	rollupAddress := ethCmn.HexToAddress(config.GlobalCfg.RollupAddress)
+	stakeAmount := big.NewInt(0)
+
+	// TODO fix
```

Comment:
What's the thing we want to fix here? Is it adding a feeReceiver to the config?

Reply:
Right now I have added 0 as fee receiver to fix build but we need to replace that with the correct one.

Lets talk here: https://github.com/thehubbleproject/hubble-commander/issues/47